File: programmers/js/new/level2/영어_끝말잇기.js
function solution(n, words) {
    const countTurn = Array(n + 1).fill(0);
    const set = new Set();
    if (words[0].length <= 1) {
        return [1, 1];
    }
    set.add(words[0]);
    countTurn[1]++;
    for (let i = 1; i < words.length; ++i) {
        const word = words[i];
        const pNum = (i % n) + 1;
        countTurn[pNum]++;
        if (set.has(word) || words[i - 1][words[i - 1].length - 1] !== word[0]) {
            return [pNum, countTurn[pNum]];
        }
        set.add(word);
    }
    return [0, 0];
}

File: programmers/python/level2/영어_끝말잇기.py
def solution(n, words):
    word_set = set()
    last_word = words[0][0]
    for i, word in enumerate(words):
        if word in word_set or last_word[-1] != word[0]:
            turn = i // n + 1
            person = i % n + 1
            return [person, turn]
        else:
            last_word = word
            word_set.add(word)
    return [0,0]